src/parser/shaman/enhancement/modules/talents/Stormflurry.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS/index';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent, DamageEvent } from 'parser/core/Events';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import ItemDamageDone from 'interface/ItemDamageDone';
+import { formatPercentage } from 'common/format';
+import { STORMSTRIKE_CAST_SPELLS, STORMSTRIKE_DAMAGE_SPELLS } from 'parser/shaman/enhancement/constants';
+
+const MAIN_HAND_DAMAGES = [
+  SPELLS.STORMSTRIKE_DAMAGE.id,
+  SPELLS.WINDSTRIKE_DAMAGE.id,
+];
+
+const OFF_HAND_DAMAGES = [
+  SPELLS.STORMSTRIKE_DAMAGE_OFFHAND.id,
+  SPELLS.WINDSTRIKE_DAMAGE_OFFHAND.id,
+];
+
+/**
+ * Stormstrike has a 25% chance to strike the target an additional time for
+ * 40% of normal damage. This effect can chain off of itself.
+ *
+ * Example Log:
+ *
+ */
+class Stormflurry extends Analyzer {
+  protected mainHandHits: number[] = [];
+  protected offHandHits: number[] = [];
+  protected totalStormStrikeCasts: number = 0;
+  protected totalStormStrikeHits: number = 0;
+  protected extraHits: number = 0;
+  protected extraDamage: number = 0;
+  protected processed: boolean = false;
+
+
+  constructor(options: any) {
+    super(options);
+    if (!this.selectedCombatant.hasTalent(SPELLS.STORMFLURRY_TALENT.id)) {
+      this.active = false;
+      return;
+    }
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER)
+        .spell(STORMSTRIKE_CAST_SPELLS),
+      this.onStormstrikeCast,
+    );
+
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER)
+        .spell(STORMSTRIKE_DAMAGE_SPELLS),
+      this.onStormstrikeDamage,
+    );
+  }
+
+  onStormstrikeCast(event: CastEvent): void {
+    this.totalStormStrikeCasts += 1;
+  }
+
+  onStormstrikeDamage(event: DamageEvent): void {
+
+    const dmg = event.amount + (event.absorbed || 0);
+
+    if (MAIN_HAND_DAMAGES.includes(event.ability.guid)) {
+      this.totalStormStrikeHits += 1;
+      this.mainHandHits.push(dmg);
+    } else if (OFF_HAND_DAMAGES.includes(event.ability.guid)) {
+      this.offHandHits.push(dmg);
+    }
+  }
+
+  determineExtraHitsAndDamage(): void {
+    this.mainHandHits.sort((a,b) => a - b);
+    this.offHandHits.sort((a, b) => a - b);
+
+    this.extraHits = this.totalStormStrikeHits - this.totalStormStrikeCasts;
+
+    const extraMainHandHits = this.mainHandHits.slice(0, this.extraHits);
+    const extraOffHandHits = this.offHandHits.slice(0, this.extraHits);
+    const extraMainHandDamage = extraMainHandHits.reduce((a, b) => a + b, 0);
+    const extraOffHandDamage = extraOffHandHits.reduce((a, b) => a + b, 0);
+
```

Comment:
If I am understanding this correctly this wont work in all circumstances.
As far as I can tell you are collecting all Mainhand/Offhand-Hits of Storm-/Windstrike, sorting them by the damage amount and assume that the ones doing the least damage are the duplicates/flurries.

The problem with it being only 40% is that a crit is still doing at least 80% of the normal damage. So the moment you get buffs that increase your damage by about 20% your sorting is not gonna work. Even a strombringer proc does this.

I'd personally go for a "window" sort of thing, like all stormstrike damage events that happen at most 0.5 seconds after another stormstrike are flurries.

Reply:
Yeah I struggled a bit with how to implement this. Figured just collecting all the baby damage would give a low estimate.
Not sure how I'd approach the window -- just track the timestamp of the last SS and compare to the current? 🤔

I'll give it a shot.

Reply:
there is some new functionality that can help with this, if im understanding what you are looking for

this.eventHistory.last(1, 500, Events.damage.by(SELECTED_PLAYER).spell(SPELLS.STORMFLURRY))

So the way this works, is (assuming both hits are the same Spell ID), if you do that on each damage hit, it will grab the last stormflurry damage event if there is one within 500ms.

So if it is the first hit, it wont find one. If it is the second hit, it should find the first one.